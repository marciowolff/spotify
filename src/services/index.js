import React from 'react';
import createClient from 'spotify-service';

import { API_URL } from '../constants';

export const services = createClient(API_URL[process.env.ENV]);
export { default as auth } from 'spotify-service';

export function withServices(Component) {
  const ComponentWithServices = props => (
    <Component service={services} {...props} />
  );

  return ComponentWithServices;
}
