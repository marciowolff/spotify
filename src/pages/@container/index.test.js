import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Container from './index';

import { auth } from '../../services';

const history = { push: jest.fn() };
const location = { pathname: '/teste' };
const spyisAuthenticated = jest.spyOn(auth, 'isAuthenticated');

let wrapper;

it('Thumb should be a function', () => {
  expect(typeof Container).toBe('function');
});

it('Should Search to match snapshot', () => {
  const tree = renderer
    .create(
      <Container location={location} history={history}>
        conteudo
      </Container>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('is authenticated', () => {
  it('Should call auth.isAuthenticated return true', () => {
    spyisAuthenticated.mockImplementation(() => true);

    wrapper = mount(
      <Container location={location} history={history}>
        conteudo
      </Container>
    );

    expect(spyisAuthenticated).toHaveBeenCalled();
    expect(history.push).not.toHaveBeenCalled();
    expect(wrapper.props().children).toEqual('conteudo');
  });

  it('Should call auth.isAuthenticated return false and location equal "/"', () => {
    spyisAuthenticated.mockImplementation(() => false);

    wrapper = mount(
      <Container location={{ pathname: '/' }} history={history}>
        conteudo
      </Container>
    );
    expect(spyisAuthenticated).toHaveBeenCalled();
    expect(history.push).not.toHaveBeenCalled();
    expect(wrapper.props().children).toEqual('conteudo');
  });

  it('Should call auth.isAuthenticated return false and location diferent "/"', () => {
    spyisAuthenticated.mockImplementation(() => false);

    wrapper = mount(
      <Container location={{ pathname: '/teste' }} history={history}>
        conteudo
      </Container>
    );
    expect(spyisAuthenticated).toHaveBeenCalled();
    expect(history.push).toHaveBeenCalledWith('/');
  });
});
