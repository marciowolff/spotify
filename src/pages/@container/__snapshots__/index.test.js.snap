// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Search to match snapshot 1`] = `
Array [
  <header
    className="page-header"
  >
    <img
      id="logo"
      src="mock-file"
    />
  </header>,
  <main
    className="container undefined"
  >
    conteudo
  </main>,
]
`;
